### DISCUSSION TITLE
What is the ccdeploy command for?

### DISCUSSION THREAD

**Discussion by iainelder** [aws-cloudformation/rain#256] [Q&A] [answered]
Via the community reviews thread I learned about the `ccdeploy` command to deploy a local template using the Cloud Control API.

How is that different from using the good old CloudFormation API to control a stack via a template?

When would you want to use Cloud Control for that instead of CloudFormation?

**Comment by ericzbeard** [comment, marked as answer]
The `cc deploy` command is experimental and not something I'd expect anyone to actually use any time soon. It's a proof of concept for deploying templates via API calls into the Cloud Control API (CCAPI). This means there is no stack, and you won't see any evidence of it in the CloudFormation console. It's closer to the experience you would get using Terraform and the `awscc` provider (although not nearly as stable or complete). With this command, Rain itself is handling a lot of what goes on behind the scenes when CloudFormation deploys a stack, so it's a good exercise to see what's involved with resource management.

**Reply by iainelder** [reply]
Thanks for the explanation.

I just learned about the [awscc](https://registry.terraform.io/providers/hashicorp/awscc/latest) provider for Terraform.

Now it's clearer to me why CloudControl exists as a separate public service. It makes it easier to run other infrastructure-as-code tools such as Terraform, Pulumi, and the rest.
